### PR TITLE
Fetch and filter all samples from a study when using a custom case list

### PR DESCRIPTION
**Pull request by adamabeshouse**
Fixes https://github.com/cBioPortal/cbioportal/issues/8593

### CODE REVIEW

**Review by Luke-Sikina — APPROVED** [review]

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -3647,32 +3648,45 @@ export class ResultsViewPageStore {
...
+                                    sampleListIds: customSampleListStudyIds.map(
+                                        studyId => `${studyId}_all`
+                                    ),
+                                } as SampleFilter,
+                                projection:
+                                    REQUEST_ARG_ENUM.PROJECTION_DETAILED,
+                            })
+                            .then(samples => {
+                                return samples.filter(s =>
+                                    customSampleListIds.has(s, [
```
Comment (alisman) [inline src/pages/resultsView/ResultsViewPageStore.ts:3684]: this could be a fuckload of samples, i take it that .has is pretty optimal?

Comment (alisman) [reply src/pages/resultsView/ResultsViewPageStore.ts:3684]: i always forget, why don't these guys have unique ids that we made on server specifically for this reason?  maybe we stopped doing that?

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts:3684]: they do, it's that the custom list specification is specified by sample id and study id, not by unique id, so we won't have them here for filtering

Comment (adamabeshouse) [reply src/pages/resultsView/ResultsViewPageStore.ts:3684]: Im going to do benchmarking to see how fast it is compared to a more specific solution for this

**Review by alisman — APPROVED** [review]

**Pull request merged**